src/main/java/appeng/parts/automation/UpgradeInventory.java

```diff
@@ -169,14 +145,43 @@ public void saveChanges()
 	{
 		this.parent.saveChanges();
 	}
-
+	
 	@Override
-	public void onChangeInventory( final IInventory inv, final int slot, final InvOperation mc, final ItemStack removedStack, final ItemStack newStack )
+	public void onChangeInventory( final IItemHandlerModifiable inv, final int slot, final InvOperation mc, final ItemStack removedStack, final ItemStack newStack )
 	{
 		this.cached = false;
 		if( this.parent != null && Platform.isServer() )
 		{
 			this.parent.onChangeInventory( inv, slot, mc, removedStack, newStack );
 		}
 	}
+	
+	private class UpgradeInvFilter implements IAEItemFilter
+	{
+
+		@Override
+		public boolean allowExtract(IItemHandlerModifiable inv, int slot, int amount) 
+		{
+			return true;
+		}
+
+		@Override
+		public boolean allowInsert(IItemHandlerModifiable inv, int slot, ItemStack itemstack) 
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)